src/sysusers/sysusers.c

```diff
@@ -84,6 +84,9 @@ static uid_t search_uid = UID_INVALID;
 static UidRange *uid_range = NULL;
 static unsigned n_uid_range = 0;
 
+static UGIDAllocationRange login_defs = {};
+static bool login_defs_need_warning = false;
```

Comment:
i wished we'd lose global vars, no add more... we should wrap the whole shebang here in a "Context" or "Manager" structure one of those days...